src/Microsoft.Net.Http.Headers/HeaderUtilities.cs

```diff
@@ -13,7 +13,7 @@ public static class HeaderUtilities
         private const string QualityName = "q";
         internal const string BytesUnit = "bytes";
 
-        internal static void SetQuality(ICollection<NameValueHeaderValue> parameters, double? value)
+        internal static void SetQuality(ObjectCollection<NameValueHeaderValue> parameters, double? value)
```

Comment:
The main pivot here is around what to do in this PR with `ObjectCollection<T>`. The main issue is that `ICollection<T>` is a **loser interface** and has no friends.

We could do something else simple here like **inherit** from `Collection<T>` and pass this around as `IList<T>`. Then we could use the indexer to avoid allocations. That would remove an object allocation per `ObjectCollection<T>` that we created.

I really just kind of #YOLOed this change because `ObjectCollection<T>` is internal. Let's discuss more tomorrow if needed.